File: Task3/ex3Kotlin/palindrom.kt
import kotlin.math.abs

// Функция проверки, является ли число трехзначным палиндромом
fun isThreeDigitPalindrome(num: Int): Boolean {
    val absoluteNum = abs(num)
    if (absoluteNum < 100 || absoluteNum > 999) return false
    return (absoluteNum / 100) == (absoluteNum % 10) // Сравниваем первую и последнюю цифру
}

fun main() {
    print("Введите числа через пробел: ")
    val input = readLine() ?: ""
    
    val results = mutableListOf<String>()
    var firstOutput = true
    
    // Обработка каждого числа из ввода
    input.split(" ").forEach { numStr ->
        try {
            val num = numStr.toInt()
            if (abs(num) in 100..999) { // Проверка на трехзначность
                val result = if (isThreeDigitPalindrome(num)) "YES" else "NO"
                if (!firstOutput) results.add(", ")
                results.add("$num - $result")
                firstOutput = false
            }
        } catch (e: NumberFormatException) {
            // Игнорируем нечисловые вводы
        }
    }
    
    // Вывод результатов
    print("Результат: ")
    results.forEach { print(it) }
}

//java -jar palindrom.jar 

File: Task3/ex3C#/palindrom.cs
using System;

class Program
{
    // Проверяет, является ли число трехзначным палиндромом
    static bool IsThreeDigitPalindrome(int num)
    {
        num = Math.Abs(num); // Работаем с абсолютным значением числа
        if (num < 100 || num > 999) return false; // Проверка на трехзначность
        return (num / 100) == (num % 10); // Сравниваем первую и последнюю цифру
    }

    static void Main()
    {
        Console.Write("Введите числа через пробел: ");
        string input = Console.ReadLine(); // Чтение ввода пользователя
        string[] numbers = input.Split(' '); // Разделение ввода на массив чисел
        
        Console.Write("Результат: ");
        bool firstOutput = true; // Флаг для форматирования вывода
        
        foreach (string numStr in numbers)
        {
            if (int.TryParse(numStr, out int num)) // Парсинг строки в число
            {
                if (Math.Abs(num) >= 100 && Math.Abs(num) <= 999) // Проверка диапазона
                {
                    if (!firstOutput) Console.Write(", "); // Добавляем запятую перед элементами, кроме первого
                    // Вывод числа и результата проверки (YES/NO)
                    Console.Write($"{num} - {(IsThreeDigitPalindrome(num) ? "YES" : "NO")}");
                    firstOutput = false; // После первого вывода сбрасываем флаг
                }
            }
        }
    }
}

//mono palindrom.exe